src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -3161,6 +3711,32 @@ namespace Microsoft.FSharp.Core
         /// <param name="action">The action to perform during the lock.</param>
         ///
         /// <returns>The resulting value.</returns>
+        /// 
+        /// <example id="lock-example">
+        /// <code lang="fsharp">
+        /// type TestCounter () =
+        ///     let mutable count = 0
+        ///     member this.PlainOldIncrement() = count <- count + 1
+        ///     member this.ThreadSafeIncrement() = lock this (this.PlainOldIncrement)
+        ///     member this.Count = count
+        ///         
+        /// open System.Linq
+        /// let counter = TestCounter()
+        /// //  Create a parallel sequence to that uses all our CPUs
+        /// ({1..100000}).AsParallel()
```

Comment:
Generally I'd always encourage people to write `seq { 1..100000 }` rather than without the `seq`